### DISCUSSION TITLE
No suitable solver in yalmip

### DISCUSSION THREAD

**Discussion by KarryMa** [yalmip/YALMIP#1192] [Problems using YALMIP] [answered]
I use two variables product in the objective function, I think that's why I get "No suitable solver " in the result.
My codes are:
```
function [X,v] = ym_Solve_FP_rhoE(n,tau1,tau2)

d = 2^n;
yalmip('clear')

                % Define variables
                X = sdpvar(d,d,'hermitian','complex');   
                v = sdpvar(d^2,d^2);
                % Define constraints
                Constraints = [X >= 0;v(2,2)+v(4,1)*2^n==1;v(1,1)==v(2,2)...
                    +v(4,1);v>=0;v(2,2)==v(3,3);v(1,1)==v(4,4)]; 
                % Define an objective
                %%% 
                Objective = 0.5*norm(y-Pauli*v*vec(X),2)^2 + tau1*trace(X)+ tau2*norm(v,1);
                sol = optimize(Constraints,Objective);
                % Analyze error flags
if sol.problem == 0
 % Extract and display value
X = value(X);
v = value(v);
else
 disp('Hmm, something went wrong!');
 sol.info
 yalmiperror(sol.problem)
end          

end
```
My result: "    'No suitable solver'"

**Comment by johanlofberg** [comment, marked as answer]
You have a nonlinear nonconvex SDP (since the objective is of fourth order) and thus there is no way to solve this as there are no solvers for that

**Reply by KarryMa** [reply]
Thank you first, but which term is nonlinear nonconvex SDP? And how can I solve this problem?

**Reply by KarryMa** [reply]
the reason is I used v*vec(X)?

**Reply by johanlofberg** [reply]
(y-v*x)^2 is quartic and nonconvex. there is no way to solve it without developing your own solver/heuristic

btw, your line break in the constraints in dangerous coding pattern as it evaluates to two terms which is not what you want most likely

```
>> [1 == 0....
    +5]

ans =

     0     5
```

**Reply by KarryMa** [reply]
Thanks a lot! And I can't use `norm(y-Pauli*v*vec(X),2)^2` in cvx package, either, right?

**Reply by johanlofberg** [reply]
Absolutely not. cvx is only for limited convex models while YALMIP supports basically anything (but here you don't have any solver, so that does not help). Well, technically the development version of YALMIP has a general nonlinear SDP framework, but it fails on this model and I would not waste energy trying to work with this model but instead devote energy or deriving linear or at most mixed-integer SDPs
